pyanaconda/installation.py

```diff
@@ -392,8 +393,9 @@ def _prepare_installation(self, payload, ksdata):
 
         # Make name resolution work for rpm scripts in chroot.
         # Also make sure dns resolution works in %post scripts
-        # if systemd-resolved is not available.
-        if conf.system.provides_resolver_config:
+        # when systemd-resolved is not available.
+        if conf.system.provides_resolver_config and \
+                not is_service_installed("systemd-resolved.service"):
```

Comment:
Dont you need here, root=self._sysroot argument? You care about the target system service, no?

Reply:
We care about installer environment service.
At this point the target system is empty and our only indication that systemd-resolved is not used on the system (and will not be installed) is that it is missing in the image ( https://github.com/rhinstaller/anaconda/pull/5488).